From e4b268180159559b768065025c5608d64c8ce75f Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Thu, 9 Jul 2020 00:04:22 +0530
Subject: [PATCH 03/25] gpoll: UWP does not need to poll a msg_fd

msg_fd is only set when we need to handle win32 Window Messages, which
are not available when building with UWP.
---
 glib/gpoll.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/glib/gpoll.c b/glib/gpoll.c
index 5afc759..148f551 100644
--- a/glib/gpoll.c
+++ b/glib/gpoll.c
@@ -151,8 +151,15 @@ poll_rest (GPollFD *msg_fd,
       if (_g_main_poll_debug)
 	g_print ("  MsgWaitForMultipleObjectsEx(%d, %lu)\n", nhandles, timeout_ms);
 
+#ifdef G_WINAPI_ONLY_APP
+      /* msg_fd is only set when we need to handle win32 Window Messages, which
+       * are not available when building with UWP
+       */
+      g_assert_not_reached ();
+#else
       ready = MsgWaitForMultipleObjectsEx (nhandles, handles, timeout_ms,
 					   QS_ALLINPUT, MWMO_ALERTABLE);
+#endif
 
       if (ready == WAIT_FAILED)
 	{
@@ -449,8 +456,15 @@ g_poll (GPollFD *fds,
 
   /* Wait for at least one thread to return */
   if (msg_fd != NULL)
+#ifdef G_WINAPI_ONLY_APP
+    /* msg_fd is only set when we need to handle win32 Window Messages, which
+     * are not available when building with UWP
+     */
+    g_assert_not_reached ();
+#else
     ready = MsgWaitForMultipleObjectsEx (nthreads, thread_handles, timeout,
                                          QS_ALLINPUT, MWMO_ALERTABLE);
+#endif
   else
     ready = WaitForMultipleObjects (nthreads, thread_handles, FALSE, timeout);
 
-- 
2.38.1

